src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int numberFigure = new Random().nextInt(4);
+        switch (numberFigure) {
+            case 0:
+                return new Circle(getRandomDouble(), colorSupplier.getRandomColor());
```

Comment:
make a method to create each figure 
```suggestion
                return getRandomCircle();
```

Reply:
why create for each figure by method? if you can just assign values?